rfcs/0002-branch-generation.md

```diff
@@ -0,0 +1,76 @@
+    a. Votes are signatures of the hash of the computed state at that PoH tick count
+    b. Votes are also propagated via gossip
+10. Leader executes the votes as any other transaction and broadcasts them to the network
+11. Validators observe their votes and all the votes from the network.
+
+## Partitions, Forks
+
+Forks can arise at PoH tick counts that correspond to a vote.  The next leader may not have observed the last vote period and may start their slot with generated virtual PoH entries.  These empty ticks are generated by all nodes in the network at a network-specified rate for hashes/per/tick `Z`.
+
+There are only two possible versions of the PoH during a voting period: PoH with `T` ticks and entries generated by the current leader, or PoH with just ticks.  The "just ticks" version of the PoH can be thought of as a virtual ledger, one that all nodes in the network can derive from the last tick in the previous period.
+
+Validators can ignore forks at other points (e.g. from the wrong leader), or slash the leader responsible for the fork.
+
+Validators vote based on a greedy choise to maximze their reward described in [branch selection](rfcs/0008-branch_selection.md).
+
+### Validator's View
+
+#### Time Progression
+The diagram below represents a validator's view of the PoH stream with possible forks over time.  L1, L2, etc. are leader periods, and `E`s represent entries from that leader during that leader's period.  The 'x's represent ticks only, and time flows downwards in the diagram.
+
+
+```
+time  +----+                                          validator action
+|     | L1 |                 E(L1)
+|     |----|            /             \                vote(E(L2))
+|     | L2 |        E(L2)               x
+|     |----|        /    \           /     \           vote(E(L2))
+|     | L3 |    E(L3)      x       E(L3)'    x
+|     |----|    /  \     /   \     /  \     /  \       slash(L3)
+|     | L4 |  x     x  E(L4)  x   x    x   x    x
+V     |----|  |     |  |      |   |    |   |    |      vote(E(L4))
+V     | L5 |  xx   xx  xx   E(L5) xx   xx  xx   xx
+V     +----+                                           hang on to E(L4) and E(L5) for more...
+
+```
+
+Note that an `E` appearing on 2 branches at the same period is a slashable condition, so a validator observing `E(L3)` and `E(L3)'` can slash L3 and safely choose `x` for that period.  Once a validator commits to a branch, other branches can be discarded below that tick count.  For any period, validators need only consider a single "has entries" chain or a "ticks only" chain to be proposed by a leader.  But multiple virtual entries may overlap as they link back to the a previous period.
+
+#### Time Division
+
+It's useful to consider leader rotation over PoH tick count as time division of the job of encoding state for the network.  The following table presents the above tree of forks as a time-divided ledger.
+
+leader period |  L1 | L2 | L3 | L4 | L5
```

Comment:
Same here, copy the table from `book/src/leader-rotation.md`. I'll watch for changes until the implementation is done and then delete the RFC, so there's no longer the duplication.